tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -29,18 +54,36 @@ func (r *RealDependencies) execute(cmd string, redirectStdout bool, args ...stri
 	fmt.Printf("%s %s\n", cmd, strings.Join(args, " "))
 	externalCommand := exec.Command(cmd, args...)
 	externalCommand.Stdout = os.Stdout
+
+	var stderr bytes.Buffer
 	// TODO Check naming and redirection logic
 	if !redirectStdout {
-		externalCommand.Stderr = os.Stderr
+		externalCommand.Stderr = &stderr
 	}
-	return externalCommand.Run()
+
+	err := externalCommand.Run()
+	stderrStr := stderr.String()
+	if err != nil && !redirectStdout {
+		// Print stderr to os.Stderr.
+		fmt.Println(stderrStr)
+
+		// Parse the exit status code.
+		exitcode, _ := strconv.Atoi(strings.Trim(strings.SplitN(err.Error(), "exit status", 2)[1], " "))
+		errtypeStr := exittypeToString[xtablesExittype(exitcode)]
```

Comment:
Two problems need to be fixed here:

1. This `execute` method is not limited to running `iptables` commands. The logic here is specific to `iptables`, so please move it into a wrapper function that will be specific to `iptables`.
2. If the command returns another exit code, this panics. This is too brittle. Make `exittypeToString` a map instead of a map and handle the case when the exit code is unexpected.

Reply:
Thanks for the great comments!
Updated with a separate helper for transforming iptables-specific errors and more error handlings.